components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/correlation/CorrelationConfigManager.java

```diff
@@ -0,0 +1,205 @@
+            JSONObject responseJson = new JSONObject(responseString);
+            List<CorrelationConfigDTO> correlationConfigDTOList = new ArrayList<>();
+            JSONArray correlationComponentsArray = responseJson.getJSONArray("components");
+            for (int i = 0; i < correlationComponentsArray.length(); i++) {
+                JSONObject correlationComponentObject = correlationComponentsArray.getJSONObject(i);
+                CorrelationConfigDTO correlationConfigDTO = new CorrelationConfigDTO();
+                correlationConfigDTO.setName(correlationComponentObject.getString("name"));
+                correlationConfigDTO.setEnabled(correlationComponentObject.getString("enabled"));
+
+                List<CorrelationConfigPropertyDTO> correlationConfigPropertyDTOList = new ArrayList<>();
+                JSONArray correlationConfigPropertiesArray = correlationComponentObject.getJSONArray("properties");
+                for (int j = 0; j < correlationConfigPropertiesArray.length(); j++) {
+                    JSONObject correlationConfigPropertyObject = correlationConfigPropertiesArray.getJSONObject(j);
+                    CorrelationConfigPropertyDTO correlationConfigPropertyDTO = new CorrelationConfigPropertyDTO();
+                    correlationConfigPropertyDTO.setName(correlationConfigPropertyObject.getString("name"));
+
+                    List<String> propertyValueList = new ArrayList<>();
+                    JSONArray propertyValueArray = correlationConfigPropertyObject.getJSONArray("value");
+                    for (int k = 0; k < propertyValueArray.length(); k++) {
+                        String propertyValue = propertyValueArray.getString(k);
+                        propertyValueList.add(propertyValue);
+                    }
+                    correlationConfigPropertyDTO.setValue(
+                            propertyValueList.toArray(new String[0]));
+                    if (correlationConfigPropertyDTO.getName().equals(DENIED_THREADS)) {
+                        deniedThreads = correlationConfigPropertyDTO.getValue();
+                    }
+                    correlationConfigPropertyDTOList.add(correlationConfigPropertyDTO);
+                }
+                correlationConfigDTO.setProperties(correlationConfigPropertyDTOList);
+                correlationConfigDTOList.add(correlationConfigDTO);
+            }
+            log.debug("Updating Correlation Config in the gateway Start Up");
+            updateCorrelationConfigs(correlationConfigDTOList);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Response : " + responseString);
+            }
+        } catch (IOException | APIManagementException e) {
+            log.error("Error while calling internal service API for correlation configs", e);
+        }
+    }
+
+    public void updateCorrelationConfigs(List<CorrelationConfigDTO> correlationConfigDTOList) {
+
```

Comment:
No need to have this empty line in the method body

Reply:
> No need to have this empty line in the method body

These changes were authored and committed by @pasindu-sandima. I will send a separate PR adding these requested changes since this PR is already merged.